File: backend-play/app/com/github/brane08/pagila/store/entities/Store.java
package com.github.brane08.pagila.store.entities;

import com.github.brane08.pagila.seedworks.entities.BaseModel;

import javax.persistence.*;
import java.util.Objects;

@Entity
@Table(name = "store")
public class Store extends BaseModel {

	@Id
	@Column(name = "store_id")
	public Integer storeId;

	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "manager_staff_id")
	public Staff manager;

	@OneToOne(optional = false)
	@JoinColumn(name = "address_id")
	public Address address;

	@Override
	public boolean equals(Object o) {
		if (this == o) return true;
		if (o == null || getClass() != o.getClass()) return false;
		Store that = (Store) o;
		return storeId.equals(that.storeId);
	}

	@Override
	public int hashCode() {
		return Objects.hash(storeId);
	}
}

File: backend-micronaut/src/main/kotlin/com/github/brane08/pagila/models/Store.kt
package com.github.brane08.pagila.models

import javax.persistence.*

@Entity
@Table(name = "store")
class Store : BaseModel() {
	@Id
	@Column(name = "store_id")
	var storeId: Int? = null

	@ManyToOne
	@JoinColumn(name = "manager_staff_id")
	var manager: Staff? = null

	@OneToOne(optional = false)
	@JoinColumn(name = "address_id")
	var address: Address? = null

	@OneToMany(mappedBy = "store")
	var currentStaff: List<Staff> = mutableListOf()
}